portals/application/modules/provision/controllers/IndexController.php

```diff
@@ -118,28 +118,63 @@ protected function _searchGenericPattern(\IvozProvider\Mapper\Sql\TerminalModels
      */
     protected function _getTerminalByUrl($terminalUrl)
     {
-        $fileName = basename($terminalUrl);
-        $fileExtensionPosition = strrpos($fileName, '.');
-        if ($fileExtensionPosition) {
-            $fileName = substr($fileName, 0, $fileExtensionPosition);
-        }
+        $fileName = $this->extractFileNameWithoutExtension($terminalUrl);
+        // We only allow a-f0-9 for macs in database
+        // ensure that format in results
+        $fileName = preg_replace('/[^0-9a-f]/i', '', $fileName);
 
         if (empty($fileName)) {
             return null;
         }
 
         $terminalMapper = new \IvozProvider\Mapper\Sql\Terminals();
+        $macRegExp = '/(?=(([0-9a-f]{2}[:-]{0,1}){5}([0-9a-f]){2}))/i';
+        preg_match_all($macRegExp, $fileName, $matches);
+        $macCandidates = array_key_exists(1, $matches)
+            ? $matches[1]
+            : null;
+
+        if (empty($macCandidates)) {
+            return null;
+        }
 
         /**
-         * @var IvozProvider\Model\Terminals $terminal
+         * @var IvozProvider\Model\Terminals[] $terminals
          */
-        $terminal = $terminalMapper->findOneByField('mac', $fileName);
+        $terminals = $terminalMapper->fetchList('mac in ("'. implode('","', $macCandidates) .'")');
+        foreach ($terminals as $candidate) {
+
+            /**
+             * @var IvozProvider\Model\TerminalModels $terminalModel
+             */
+            $terminalModel = $candidate->getTerminalModel();
+            if (!$terminalModel) {
+                continue;
+            }
 
-        if (!$terminal) {
-            return null;
+            $especificUrl = $this->extractFileNameWithoutExtension(
+                $terminalModel->getSpecificUrlPattern()
+            );
+            $fixedEspecificUrl = str_ireplace('{mac}', $candidate->getMac(), $especificUrl);
+
+            if (strtolower($fixedEspecificUrl) === strtolower($fileName)) {
+                return $candidate;
+            }
+        }
+
+        return null;
+    }
+
+    protected function extractFileNameWithoutExtension($route)
```

Comment:
We did this before for multimedia worker using pathinfo. It works with strings (the file doesn't have to exist), check it out [here](https://github.com/irontec/ivozprovider/blob/bleeding/portals/application/workers/MultimediaWorker.php#L50)